mlxtend/frequent_patterns/apriori.py

```diff
@@ -122,50 +141,80 @@ def apriori(df, min_support=0.5, use_colnames=False, max_len=None, verbose=0):
                              '`df.columns = [str(i) for i in df.columns`].')
 
         X = df.to_coo().tocsc()
-        support = np.array(np.sum(X, axis=0) / float(X.shape[0])).reshape(-1)
+        support = _support(X, X.shape[0], is_sparse)
     else:
         X = df.values
-        support = (np.sum(X, axis=0) / float(X.shape[0]))
+        support = _support(X, X.shape[0], is_sparse)
 
     ary_col_idx = np.arange(X.shape[1])
     support_dict = {1: support[support >= min_support]}
     itemset_dict = {1: ary_col_idx[support >= min_support].reshape(-1, 1)}
     max_itemset = 1
     rows_count = float(X.shape[0])
 
-    if max_len is None:
-        max_len = float('inf')
-
     iter_count = 0
+    all_ones = np.ones((int(rows_count), 1))
 
-    while max_itemset and max_itemset < max_len:
+    while max_itemset and max_itemset < (max_len or float('inf')):
         next_max_itemset = max_itemset + 1
         combin = generate_new_combinations(itemset_dict[max_itemset])
-        frequent_items = []
-        frequent_items_support = []
-
-        if is_sparse:
-            all_ones = np.ones((X.shape[0], next_max_itemset))
-        for c in combin:
-            if verbose:
-                iter_count += 1
-                print('\rIteration: %d | Sampling itemset size %d' %
-                      (iter_count, next_max_itemset), end="")
+        if not low_memory:
+            combin = np.array(list(combin))
+
+            if combin.size == 0:
+                break
+
+        if verbose and not low_memory:
+            print('\rProcessing %d combinations | Sampling itemset size %d' %
+                  (combin.size, next_max_itemset), end="")
+
+        # With exceptionally large datasets, the matrix operations can use a
+        # substantial amount of memory. For low memory applications or large
+        # datasets, set low_memory = True to use a slower but more memory-
+        # efficient implementation.
+        if low_memory:
```

Comment:
I think the code would be a little bit more efficient if 

```python
if not low_memory:
...
if low_memory:
```

was changed to

```python
if not low_memory:
...
else:
```

Reply:
Agreed, that was sloppy. Fixed!

Reply:
thanks!